index.rst

```diff
@@ -638,12 +698,92 @@ The following require different procedures to create/modify a configuration
 
 - :ref:`Main and Auxiliary Telescope Pointing Components <section-pointing-component>`
 - :ref:`ATMCS and ATPneumatics <section-atmcs-atpneumatics>`
+- Camera CSCs
+
+CSC Developer Information
+=========================
+
+This section contains information that is primarily of interest to CSC developers.
+However, if issues are being encountered when creating new configurations the information may be pertinent.
+
+Rules Regarding Configuration Definitions and Usage
+---------------------------------------------------
+#.  No default values shall be contained in the configuration schema definition in the CSC repository.
+
+#.  In the configuration repository for the given CSC (e.g `ts_config_attcs <https://github.com/lsst-ts/ts_config_attcs>`_ for the ATDome) there shall be a ``_init.yaml`` file that specifies values that are expected to be common to all sites and/or be relatively static in operations (we intentionally use "_init" instead of "_default").
+
+    - See this :download:`example _init.yaml <_static/_init.yaml>` for the ATSpectrograph CSC.
+    - This file is the first configuration file loaded by the CSC.
+    - Providing the ``_init.yaml`` file (or any file with a ``_`` prefix) to the ``configurationOverride`` parameter will return an error.
+    - Note that all CSCs having multiple algorithms [2]_, each with different required configuration parameters, must have an initial set of defaults in this file.
```

Comment:
I'm not sure what you are getting at. My guess is that you are trying to say that a CSC with pluggable algorithms must have a default that is usable (i.e. specifies suitable algorithm(s) and their necessary configuratIon).

Providing an example could help. The Watcher is probably the poster child for pluggable algorithms.

Also: unlike the main CSC configuration schema, algorithm configuration schemas are allowed to contain default values. (Trying to configure something like the Watcher would be a complete nightmare otherwise.)

I think it is well worth pointing that out here!

Reply:
Maybe @tribeiro should make a suggestion here. I believe it's his original text and I'm not very familiar with the CSCs having multiple algorithms.

Reply:
ATDomeTrajectory and MTDomeTrajectory are, I think, the original CSCs that have pluggable algorithms, but both of these packages only have one algorithm each (we have deferred work on making fancier algorithms). Watcher is entirely built on pluggable algorithms, in this case "rules".

Reply:
I think this is trying to account for "conditional schema", which is a pretty rare and advanced feature. I don't think neither the Watcher, ATDomeTrajectory and MTDomeTrajectory are in this category (at least not in their current versions).

The Watcher is actually a different case from the other two, where the rule schema is in the rules themselves and are not part of the CSC schema.

I think we can still require all CSCs to provide defaults the same way, hence my suggestion is to remove this phrase.